### PR TITLE
Logging & parameterization improvements

### LINKED ISSUE #11: Handle single result case
As pointed out by Ohmmmmm on discord, when the search query has only a single result it redirects straight to the record page rather than displaying a table

### PR DESCRIPTION
**Pull request by thashepherd**
The sheer verbosity of scrapy's default logging was cramping my dev style, so I changed it. This PR also includes a few other tweaks.

* Default log_level for built-in scrapy modules to WARN while retaining it at INFO for our own code (this was a PITA)
* `start_date` and `open_in_browser` may be passed on the command line: `python main.py --start_date '08/01/2023'`
* docker-compose will read mailtrap (and other) env vars from `.env.dev`. In other words, if you create `.env.dev` in your repo root with
```
MAILTRAP_API_TOKEN=<your_api_token>
MAILTRAP_SENDER_ADDRESS=mailtrap@codefordayton.org
MAILTRAP_TO_ADDRESS=<your_email>
MAILTRAP_CC_ADDRESS=
MAILTRAP_BCC_ADDRESS=
```
and then run `docker-compose up --build`, those variables will be loaded in. `.env.dev` is listed in `.gitignore` so there's no risk of accidentially committing it.
* Not supplying `MAILTRAP_CC_ADDRESS` or `MAILTRAP_BCC_ADDRESS` no longer causes an error
* `start_date` and `permit_types` are passed to Jinja, and the template's been tweaked to use them: image

Output from a run with default arguments:
```
[+] Running 1/0
 ✔ Container demolition_checker-demolition-checker-1  Recreated                                                                                                                                                                                         0.0s 
Attaching to demolition-checker-1
demolition-checker-1  | 2024-08-15 19:36:24 [DemolitionSpider] INFO: Initializing spider with permit_type Building/Wrecking/Commercial/NA and start_date 08/14/2024
demolition-checker-1  | 2024-08-15 19:36:26 [DemolitionSpider] INFO: Extracted 2 records
demolition-checker-1  | 2024-08-15 19:36:26 [DemolitionSpider] INFO: 
demolition-checker-1  |         Building/Wrecking/Commercial/NA:                                                                                                                                                                                             
demolition-checker-1  |         WRK2024C-00017 - BLADECUTTERS - 1745 WOODMAN, DAYTON OH                                                                                                                                                                      
demolition-checker-1  | 2024-08-15 19:36:26 [DemolitionSpider] INFO:                                                                                                                                                                                         
demolition-checker-1  |         Building/Wrecking/Commercial/NA:
demolition-checker-1  |         WRK2024C-00016 -  - 1745 WOODMAN DR, DAYTON OH                                                                                                                                                                               
demolition-checker-1  | 2024-08-15 19:36:26 [__main__] INFO: Spider closed: DemolitionSpider, reason: finished                                                                                                                                               
demolition-checker-1  | 2024-08-15 19:36:26 [__main__] INFO: Collected 2 records, total: 2                                                                                                                                                                   
demolition-checker-1  | 2024-08-15 19:36:26 [DemolitionSpider] INFO: Initializing spider with permit_type Building/Wrecking/Residential/NA and start_date 08/14/2024                                                                                         
demolition-checker-1  | 2024-08-15 19:36:27 [DemolitionSpider] INFO: No records found
demolition-checker-1  | 2024-08-15 19:36:27 [__main__] INFO: Spider closed: DemolitionSpider, reason: finished
demolition-checker-1  | 2024-08-15 19:36:27 [__main__] INFO: Collected 0 records, total: 2                                                                                                                                                                   
demolition-checker-1  | 2024-08-15 19:36:27 [__main__] INFO: All records: 2                                                                                                                                                                                  
demolition-checker-1 exited with code 0
```

Note that this change does NOT fix https://github.com/codefordayton/demolition_checker/issues/11. 

If you're tired of seeing tiny formatting changes in reviews like this one: 
```
pip install -r requirements-dev.txt
```
...will install pre-commit which'll run `ruff` and otherwise enforce a consistent standard.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,39 +1,5 @@
...
-six==1.16.0
-tldextract==5.1.2
-Twisted==24.3.0
-typing_extensions==4.12.2
-urllib3==2.2.2
+Twisted==24.7.0
 w3lib==2.2.1
-zope.interface==7.0.1
-mailtrap==2.0.1
-jinja2==3.0.3
```
Comment (thashepherd) [inline requirements.txt:39]: If you think this looks funny - I used https://github.com/fpgmaas/deptry to separate our direct dependencies (Scrapy, Twisted) from indirect dependencies (charset-normalizer, certifi) that were pulled in automatically. 

One (of many) of the great things about using a dependency manager like Poetry (or Hatch or Rye) is that it separates your direct dependencies in `pyproject.toml` from indirect dependencies, which are stored in a separate `poetry.lock`. While converting this project to [poetry ](https://python-poetry.org/) would be quick, I don't want to single-handedly inflict a change on your personal development process by doing so.

File: main.py [hunk 2]
```diff
@@ -6,45 +6,93 @@
...
-    main()
\ No newline at end of file
+    import argparse
+
+    parser = argparse.ArgumentParser()
+    # Note that start_date defaults to yesterday
+    parser.add_argument(
+        "--start_date",
+        required=False,
+        default=(date.today() - timedelta(days=1)).strftime("%m/%d/%Y"),
```
Comment (thashepherd) [inline main.py:87]: Change this to `weeks = 1` in order to cause the recreational spider to hit https://github.com/codefordayton/demolition_checker/issues/11

File: requirements.txt (hunk 1 above)
Comment (DavidEBest) [reply requirements.txt:39]: I'm game for switching to poetry. `pip` is just muscle memory for me at this point - I rarely even consider that package management has improved beyond the `requirements.txt` file.

Comment (thashepherd) [reply requirements.txt:39]: I'm happy to conduct that switch in a future PR if you want to see what that looks like. I don't know if it's worth sacrificing you & the other contributors' muscle memory for that on a project like this, though. Up to you!

Comment (DavidEBest) [reply requirements.txt:39]: That works for me!

**Comment by DavidEBest** [comment]
:shipit:

**Pull request merged**